.ci/build-kit/scripts/compile.sh

```diff
@@ -2,6 +2,14 @@
 
 set -e
 
+
+EVEREST_CMAKE_PATH=/usr/lib/cmake/everest-cmake
+EVEREST_CMAKE_VERSION=3aec8e3b058a9a1edb843da4fe46bb12d7aef071
```

Comment:
- note sure if adding everest-cmake to `dependencies.yaml` will work in the current implementation as the cmake specific edm function `evc_setup_edm` is defined in everest-cmake and you'll get an chicken-egg problem
- concerning `ci-workspace.yaml`, I wouldn't try to add additional coupling of the ci to edm.  First we should try to solve the problem, that the repo specific Dockerfile isn't used.

Reply:
> note sure if adding everest-cmake to dependencies.yaml will work in the current implementation 

It should work since when building with cmake it could ignore this one dependency and use the provided everest-cmake installation. But when running in ci edm can read this dependency from dependencies.yaml.

I agree that it could be confusing defining `everest-cmake` in `dependencies.yaml` without using it in the cmake build. This could improved by adding in line comment and set `cmake_condition` to `OFF`

> concerning ci-workspace.yaml, I wouldn't try to add additional coupling of the ci to edm.

Why not using edm in ci? It is the EVerest dependency manager which is used during build, but also as dev tool to manage the dev workspace. 

The advantage I see or my motivation is to have a transparent/visible way of defining dependencies. Doing it in `compile.sh` is hidden in my feeling.

Moving it into the `Dockerfile` without using edm for it would also be an option.

> First we should try to solve the problem, that the repo specific Dockerfile isn't used.

I am not aware of this issue, do you have an example or a log file?

Reply:
> I am not aware of this issue, do you have an example or a log file?

Sorry my bad. I remember, we deactivated this for everest-core since we have this issue with running this in PRs

Reply:
It might work by accident, but probably with no real well defined behavior.  I would not rely on that.
Concerning the coupling in the ci: I didn't mean that we should not run edm in the ci as it is run locally - but I wouldn't add an extra dependency overwrite file for the ci (as both the local build and the ci build should be similar).
Using `find_package(everest-cmake X.Y.Z) we can already express, that we need a newer version.

I totally agree, that moving it to the `compile.sh` was a hack to get the ci build, but it should not be committed this way.  Having it in the Dockerfile, should be the way to go - but that doesn't work currently.